fixtures/uitests/tests/ui/non_hashable_record_key.stderr

```diff
@@ -0,0 +1,7 @@
+error: Failed to generate scaffolding from UDL file at ../../../fixtures/uitests/src/records.udl
```

Comment:
This test seems to be highlighting a limitation of this implementation - there's no indication that a non-hashable key is used, so the end-user will have zero idea what's wrong here. Contrast with `interface_not_sync_and_send.stderr` where special hacks were made to ensure 'the trait `Sync` is not implemented for' appears in the output - can we do something similar there - have the scaffolding generate special code to trigger a specific message - ISTM that ` uniffi::deps::static_assertions::assert_impl_all!`, as used by that `Send+Sync` example, would also work here?

(I guess I'm fine with having an issue for this as a followup though)

Reply:
I'll look into this.

Reply:
I spent a bit on this now. It's harder to do than in the exampels you give because we don't have an explicit template where we could do the assertions.
I'm going to land this as is now though and do a deep-dive in a followup.